numba/parfors/parfor_lowering.py

```diff
@@ -393,6 +393,13 @@ def _lower_parfor_parallel(lowerer, parfor):
                     # reduction variable's name.  This fixes problems with
                     # cases where there are multiple assignments to the
                     # reduction variable in the parfor.
+                    reduction_var = scope.get_exact(name)
+                    if isinstance(inst, ir.Assign) and reduction_var.unversioned_name == inst.target.unversioned_name:
+                        lowerer.lower_inst(parfor_reddict[name][1][-1])
```

Comment:
I am still very scared of this line.  I'd rather see getting the name of the reduction variable in the scope at the end of the parfor and manually creating an assignment rather than relying on exactly that to be the last instruction in the collection set of reduction operations.

Reply:
oops, i forgot to remove this line. The manually creating assignment is there following this line.